plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,259 @@
+defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Workflow Category Transition Plugin
+ *
+ * @since  DEPLOY_VERSION
+ */
+final class Category extends CMSPlugin implements SubscriberInterface
+{
+    use WorkflowPluginTrait;
+    use DatabaseAwareTrait;
+    /**
+     * Load the language file on instantiation.
+     *
+     * @var    bool
+     * @since  DEPLOY_VERSION
+     */
+    protected $autoloadLanguage = true;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onAfterDisplay'             => 'onAfterDisplay',
+            'onContentPrepareForm'       => 'onContentPrepareForm',
+            'onWorkflowAfterTransition'  => 'onWorkflowAfterTransition',
+        ];
+    }
+
+    /**
+     * The form event.
+     *
+     * @param   PrepareFormEvent  $event  The event
+     *
+     * @since   DEPLOY_VERSION
+     */
+    public function onContentPrepareForm(PrepareFormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+
+        $context = $form->getName();
+
+        // Extend the transition form
+        if ($context === 'com_workflow.transition') {
+            $this->enhanceWorkflowTransitionForm($form, $data);
+
+            return;
+        }
+
+        if ($context === 'com_content.article') {
+            if (empty($data && $data->id)) {
+                return;
+            }
+            $this->disableCategoryField($form, $data);
+        }
+    }
+
+    /**
+     * Manipulate the generic list view
+     *
+     * @param   DisplayEvent  $event
+     *
+     * @return  void
+     *
+     * @since   4.0.0
+     */
+    public function onAfterDisplay(DisplayEvent $event)
+    {
+        if (!$this->getApplication()->isClient('administrator')) {
+            return;
+        }
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function()
+            {
+                var dropdown = document.getElementById('toolbar-status-group');
+                if (!dropdown){
+                    return;
+                }
+                var batchButton = document.getElementById('status-group-children-batch');
+                if (batchButton){
+                    batchButton.addEventListener('click', function()
+                    {
+                        var observer = new MutationObserver(function(mutations, observer) {
+                            var categorySelector = document.getElementById('batch-category-id');
+                            if (categorySelector) {
+                                categorySelector.disabled = true;
+                                observer.disconnect();
+                            }
+                        });
+
+                        observer.observe(document.body, { childList: true, subtree: true });
+                    });
+                }
+            });
+        ";
+
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
```

Comment:
The "problem" with JS is, that everybody could go in with the browser inspector and enables the JS again. So we need here a server-side stop, too.